xbmc/platform/win32/WIN32Util.cpp

```diff
@@ -1218,6 +1218,77 @@ bool CWIN32Util::SetThreadLocalLocale(bool enable /* = true */)
   return _configthreadlocale(param) != -1;
 }
 
+#ifndef TARGET_WINDOWS_STORE
+std::wstring CWIN32Util::GetCurrentDisplayName()
+{
+  MONITORINFOEXW mi{};
+  mi.cbSize = sizeof(mi);
+  if (!GetMonitorInfoW(MonitorFromWindow(g_hWnd, MONITOR_DEFAULTTOPRIMARY), &mi))
+    return {};
+  return mi.szDevice;
+}
+
+CWIN32Util::DisplayConfigSnapshot CWIN32Util::GetDisplayConfigSnapshot()
+{
+  uint32_t pathCount{0};
+  uint32_t modeCount{0};
+  CWIN32Util::DisplayConfigSnapshot snapshot{};
+
+  constexpr uint32_t flags{QDC_ONLY_ACTIVE_PATHS};
+  LONG result{ERROR_SUCCESS};
+
+  do
+  {
+    if (GetDisplayConfigBufferSizes(flags, &pathCount, &modeCount) != ERROR_SUCCESS)
+      return {};
+
+    snapshot.paths.resize(pathCount);
+    snapshot.modes.resize(modeCount);
+
+    result = QueryDisplayConfig(flags, &pathCount, snapshot.paths.data(), &modeCount,
+                                snapshot.modes.data(), nullptr);
+  } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+  if (result != ERROR_SUCCESS)
+    return {};
+
+  snapshot.paths.resize(pathCount);
+  snapshot.modes.resize(modeCount);
+
+  return snapshot;
+}
+
+std::optional<CWIN32Util::DisplayConfigId> CWIN32Util::GetCurrentDisplayTargetId()
+{
+  std::wstring name{GetCurrentDisplayName()};
+  if (name.empty())
+    return std::nullopt;
+
+  return GetDisplayTargetId(name);
+}
+
+std::optional<CWIN32Util::DisplayConfigId> CWIN32Util::GetDisplayTargetId(
+    const std::wstring& gdiDeviceName)
+{
+  const DisplayConfigSnapshot snapshot{GetDisplayConfigSnapshot()};
+
+  DISPLAYCONFIG_SOURCE_DEVICE_NAME source{};
+  source.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+  source.header.size = sizeof(source);
+
+  for (const auto& path : snapshot.paths)
```

Comment:
```suggestion
  for (const auto& path : GetDisplayConfigSnapshot().paths)
```

Not necessary snapshot variable because only is used here